resolver.go

```diff
@@ -31,13 +35,27 @@ type Resolver struct {
 	// A set of key/value pairs used to filter the result set. Only addresses of
 	// nodes that have matching metadata will be returned by LookupService.
 	NodeMeta map[string]string
+
+	// Cache used by the resolver to reduce the number of round-trips to consul.
+	// If set to nil then no cache is used.
+	//
+	// ResolverCache instances should not be shared by multiple resolvers
+	// because the cache uses the service name as a lookup key, but the resolver
+	// may apply filters based on its configuration.
+	Cache *ResolverCache
 }
 
 // LookupService resolves a service name to a list of addresses using the
-// resolver's configuration and the given list of extra service tags to narrow
-// the result set. Only addresses of healthy services are returned by the lookup
-// operation.
-func (rslv *Resolver) LookupService(ctx context.Context, name string, tags ...string) (addrs []net.Addr, err error) {
```

Comment:
I got rid of the ability to specify tags here because otherwise they should be included in the cache key.
Filtering on service tags can still be done by setting the `Resolver.ServiceTags` field.